sdk/samples/iot/CMakeLists.txt

```diff
@@ -137,6 +137,51 @@ target_link_libraries(paho_iot_hub_twin_sample
 
 create_map_file(paho_iot_hub_twin_sample paho_iot_hub_twin_sample.map)
 
+# Twin CBOR Sample - Only compile if intel/tinycbor has been found
+find_library(tinycbor tinycbor
+  if(WIN32)
+    PATHS $ENV{Path}
+    PATH_SUFFIXES tinycbor/lib lib
+  endif()
+)
+
+if (${tinycbor} STREQUAL "tinycbor-NOTFOUND")
+  message("-- Could NOT find tinycbor. Not compiling paho_iot_hub_twin_cbor_sample")
+else()
+  if(MSVC)
+    set(CMAKE_EXE_LINKER_FLAGS
+        "${CMAKE_EXE_LINKER_FLAGS} /NODEFAULTLIB:LIBCMT.lib")
```

Comment:
what is this doing?

Reply:
suppresses this warning: 
```
LINK : warning LNK4098: defaultlib 'LIBCMT' conflicts with use of other libs; use /NODEFAULTLIB:library [C:\Repos\momuno_azu
re-sdk-for-c\build\sdk\samples\iot\paho_iot_hub_twin_cbor_sample.vcxproj]
```
The cbor library uses by default a different default runtime library (the multithreaded LIMCMT.lib) than what the SDK is using (the multithreaded debug LIBCMTD.lib).
https://docs.microsoft.com/en-us/cpp/error-messages/tool-errors/linker-tools-warning-lnk4098?view=msvc-160

@ahsonkhan , I have tried to use the add_compile_options you suggested, and even attempted these three simplified versions:
`add_compile_options( /MT )`, `add_compile_options( /MTd )`, `add_compile_options( /MT /MTd)`, but the issue is that only one runtime default library can be used, not two.  If I use `/MT` I get the LIBCMTD.lib complaint.  If I use `/MTd` I get the LIBCMT.lib complaint.  It looks like I have to choose one, so will need to use the `/NODEFAULT:<lib>` on one of the options.

Reply:
@ahsonkhan side note, I'm not sure how effective this is to include multiple types of runtime libraries.  Only one can be used.
```
  add_compile_options(
    $<$<CONFIG:>:/MT> #---------|
    $<$<CONFIG:Debug>:/MTd> #---|-- Statically link the runtime libraries
    $<$<CONFIG:Release>:/MT> #--|
  )
  ```

Reply:
There are multiple libraries being linked to the samples, such as paho and openssl, beyond just the SDK compiled code.  Even when the SDK appears to have no build type set, which means it should be using /MT, it is possible one of these other libraries was compiled with Debug, which could explain the persistent /MTd complaint when I compile the CBOR sample with /MT.  I think it is simplest for the CBOR sample to suppress the /MT library requirement.

Reply:
Might be beating a dead horse here but I had added these guys here a while back for that:
https://github.com/Azure/azure-sdk-for-c/blob/eb9b39275c973874f029a986fbf9010a6be8b486/eng/cmake/global_compile_options.txt#L9-L13

Have you tried these?

Reply:
Yes, this is exactly what I had tried and it was not working.  I'm not sure if its fully effective tbh.  WIth not setting a CMAKE_BUILD_TYPE, the empty CONFIG should be getting used, but all the projects show up as /MTd, which make me wonder if Visual Stuido uses that as a default?  Not sure, and I ran out of steam trying to debug it. :/